src/types.rs

```diff
@@ -2,13 +2,95 @@
 pub type BlockHeight = u64;
 
 /// Configuration struct for NEAR Lake Framework
+/// NB! Consider using [`LakeConfigBuilder`]
+/// Building the `LakeConfig` example:
+/// ```
+/// use near_lake_framework::LakeConfigBuilder;
+/// # async fn main() {
+///    let config = LakeConfigBuilder::default()
+///        .s3_bucket_name("near-lake-data-testnet")
+///        .start_block_height(82422587)
+///        .build()
+///        .expect("Failed to build LakeConfig");
+/// # }
+/// ```
+#[derive(Default, Builder, Debug)]
 pub struct LakeConfig {
     /// AWS S3 compatible custom endpoint
+    ///
+    /// In case you want to run your own [near-lake](https://github.com/near/near-lake) instance and store data in some S3 compatible storage ([Minio](https://min.io/) or [Localstack](https://localstack.cloud/) as example)
+    ///
+    /// You can owerride default S3 API endpoint by using `s3_endpoint` option or [LakeConfigBuilder::custom_endpoint]"]
+    #[builder(setter(custom, strip_option, name = "custom_endpoint"), default)]
     pub s3_endpoint: Option<String>,
     /// AWS S3 Bucket name
+    #[builder(setter(into))]
     pub s3_bucket_name: String,
```

Comment:
I find it weird that we try to hide "s3_" in some cases (`s3_endpoint` is renamed to `custom_endpoint`), but `s3_bucket_name` is left as is. What is the point?

Reply:
The custom endpoint in most cases is not s3, but s3 compatible, I didn't find a point to mention s3 there. But I'm not sure about this decision

Reply:
`s3_compatible_endpoint`?